spec/intl.html

```diff
@@ -1338,6 +1338,101 @@ <h1>CalendarEraYear ( _calendar_, _dateLike_ )</h1>
             1. Return _result_.
           </emu-alg>
         </emu-clause>
+
+        <emu-clause id="sec-temporal-calendardatefromfields">
+          <h1>CalendarDateToISO ( _calendar_, _fields_, _overflow_ )</h1>
+          <p>The abstract operation CalendarDateToISO performs implementation-defined processing to convert _fields_, which is a Record that describes a date in the calendar represented by the string _calendar_ to the ISO calendar and returns the corresponding day, month and year values as a Record.</p>
+        </emu-clause>
+
+        <emu-clause id="sec-temporal-calendardateaddition">
+          <h1>CalendarDateAddition ( _calendar_, _date_, _duration_, _overflow_ )</h1>
+          <p>The abstract operation CalendarDateAddition performs implementation-defined processing to add _duration_ to _date_ in the context of the calendar represented by the string _calendar_ and returns the corresponding day, month and year of the result in the ISO calendar values as a Record.</p>
+        </emu-clause>
+
+        <emu-clause id="sec-temporal-calendardatedifference">
+          <h1>CalendarDateDifference ( _calendar_, _one_, _two_, _largestUnit_ )</h1>
+          <p>The abstract operation CalendarDateDifference performs implementation-defined processing to find the difference between the two dates _one_ and _two_ in the context of the calendar represented by the string _calendar_ and returns the corresponding years, months, weeks and days as a Record.</p>
+        </emu-clause>
+
+        <emu-clause id="sec-temporal-calendardateera">
+          <h1>CalendarDateEra ( _calendar_, _date_ )</h1>
+          <p>The abstract operation CalendarDateEra performs implementation-defined processing to find the era for the date corresponding to _date_ in the context of the calendar represented by the string _calendar_ and returns it.</p>
+        </emu-clause>
+
+        <emu-clause id="sec-temporal-calendardateerayear">
+          <h1>CalendarDateEraYear ( _calendar_, _date_ )</h1>
+          <p>The abstract operation CalendarDateEraYear performs implementation-defined processing to find the year of the current era for the date corresponding to _date_ in the context of the calendar represented by the string _calendar_ and returns it.</p>
```

Comment:
Given that this and other AOs are the only guidance for implementers, don't they need a lot more descriptive text about how implementations are supposed to behave?  For example:
* The year is generally 1-based (where 1 is the first year of the era) except in calendars like `buddhist` that have a year zero
* Should describe how to handle numbering before the start of the earliest era. (I assume use zero and negative numbers?
* Should discuss how to handle cases where eras start midyear, which I think is that dates after the era start are given the era year 1 (or zero if there's a year zero in that calendar) and dates before the era start are given the largest era year number from the previous era.

And so on. If there's no clear algorithm then there should be very detailed explanations of what implementers need to do!

Reply:
This is actually WIP as part of a larger initiative for adding more guidance in Intl about how to handle different calendars. I think adding it here would not be wise though, since this AO is supposed to contain information that applies across all current (and funnily, future) calendars, so only very general guidance like "this should return a string and not a number"

Reply:
The guidance I wrote above (and in other comments) is intended to apply to all calendars. It only uses examples for specific calendars to illustrate general points about how it should behave.

It's OK if that guidance goes somewhere else, but given that guidance about `eraYear` is scoped to this property and this AO, it seems appropriate to put it here?

Reply:
I agree that this guidance is necessary, but I think I agree with @ryzokuken that it is better to put it somewhere else, and keep the description here limited to the "intention" of the implementation-defined processing. It's my understanding that that's also what @FrankYFTang said in today's Temporal meeting, but I'd love to hear from Frank directly.

Reply:
> better to put it somewhere else

Where is that "somewhere else", and how will implementers find it from the spec?  Also, what is "intent" and what is "not intent"?

Reply:
I think the current consensus is to add it to the ECMA-402 spec.